### ISSUE TITLE
Feature: Add checking ignore patterns for JSX properties name in sort-object

### ISSUE THREAD

**Issue by hstemplewski** [azat-io/eslint-plugin-perfectionist#96]
### What rule do you want to change?

JSX properties are missed when checking ignore rule pattern

### Describe the problem

When we are using React as a framework and added perfectionist which sorts object keys, the pattern for ignore sorting does not check the JSX prop names when the object is passed directly to the component. 

### Code example

```ts
<Table
  //This one shouldn't be sorted
  columns={{
    firstCol: "COL 1",
    secondCol: "COL 2",
    thirdCol: "COL 3",
    fourhtCol: "COL 4"
  }}
/>
```
the rule is applied like: 
```ts
"perfectionist/sort-objects": [
      "error",
      {
        type: "natural",
        order: "asc",
        "ignore-pattern": ["[Cc]olumn*"],
        groups: ["id", "unknown"],
        "custom-groups": {
          id: "id",
        },
      },
    ],
```

### Additional comments

I think this [one needs](https://github.com/azat-io/eslint-plugin-perfectionist/blob/f894ad7d08fc1a4b3e95c0cace96adbf800027fa/rules/sort-objects.ts#L136) to be changes from 
```ts
 let variableIdentifier =
        node.parent.type === 'VariableDeclarator' &&
        node.parent.id.type === 'Identifier'
          ? node.parent.id.name
          : null

```
to (or sth simmillar)
```ts
 let variableIdentifier =
        (node.parent.type === 'VariableDeclarator' || node.parent.type === 'JSXAttribute') &&
        node.parent.id.type === 'Identifier'
          ? node.parent.id.name
          : null

```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Thanks, that's a really good point, I'll think about how this can be fixed.

**Comment by azat-io** [comment]
Thank you for your issue!

This feature was implemented in da515cebc247587a9824e6e20582c93fbd172eaa and released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054
